store/src/java/com/zimbra/cs/mailbox/MailItem.java

```diff
@@ -2703,6 +2712,10 @@ MailItem copy(Folder folder, int copyId, String copyUuid, MailItem parent) throw
      *    <li><tt>service.PERM_DENIED</tt> - if you don't have sufficient
      *        permissions</ul> */
     MailItem icopy(Folder target, int copyId, String copyUuid) throws IOException, ServiceException {
+        return icopy(target, copyId, copyUuid, 0);
+    }
+
+    MailItem icopy(Folder target, int copyId, String copyUuid, int trashFolderId) throws IOException, ServiceException {
```

Comment:
@gabhinavTI do we need trashFolderId param,  as there is already target Folder object can we not get the Id from it for comparison? If yes we won't need overloading these methods

Reply:
@dasiyogesh `target` param is available only in icopy() , hence removed overloading for this method but other methods  `finishCreation()`  and `updateSize() `overloading is still required hence using boolean values as for isTrashFolder in paramter